### DISCUSSION TITLE
Question about matching

### DISCUSSION THREAD

**Discussion by amo13** [jonaswinkler/paperless-ng#912] [Support] [answered]
I noticed that a document I dropped into paperless containing the word "Rentenversicherungsnummer" would not match the word "Rentenversicherung". For it do actually match, I have to add the full word to the match strings.
Is there a way to enter a word in the match field and allow it to be only a part of a word? I tried "Rentenversicherung*" but that did not work. 

The reason I ask is that in some cases one might not guess all possible word ending which should still trigger the same match.
As an example: I'd like to categorize a document as a "Arbeitsunfähigkeitsbescheinigung" by triggering a match. I put `arbeitsunfähig Arbeitsunfähigkeit Arbeitsunfähigkeitsbescheinigung` in the match field and then later notice that those documents are not matching because they may have a line break like this:
```
Arbeitsunfähigkeits-
bescheinigung
```
Maybe I did not foresee this, or the word was broken somewhere else. For it to match correctly, I then would need to add `Arbeits- Arbeitsunfähigkeits- Arbeitsunfähigkeitsbeschei-` to the match field. Therefore, I'd like to put something like `arbeitsunfähig*` (case insensitive) in the match field.

**Comment by amo13** [comment, marked as answer]
Ok, so I think I can answer my own question now. As a somewhat "uncomfortable" matching pattern I overlooked regular expressions. This is what I have to use in that case.
Therefore, for matching `*arbeitsunfähig*`, I can use the regular expression `arbeitsunfähig` and set it to case insensitive.
If I need to match any of multiple subwords, I can separate the subword candidates by `|` instead of spaces: `sub1|sub2|sub3`

So I guess it's good that it's possible, but my conclusion from this is that a checkbox "allow substrings" next to "case insensitive" would be great. Using regular expressions might be too much for people who are afraid of anything that looks just roughly like programming code.

**Reply by steakhutzeee** [reply]
Hi, could you please elaborate a little more how you managed to search for a part of a string?

Thanks!

**Reply by amo13** [reply]
If you mean just for searching, then it looks to me like you can select "advanced search" and just type `*substring*`. That works for me at least.
I was looking for a way to match against a substring during the matching process for assigning tags, document types and correspondents. And this, I found out, can be achieved by just entering the substring in the match field and selecting the `regular expression` instead of `any` above.

**Reply by steakhutzeee** [reply]
> 
> 
> If you mean just for searching, then it looks to me like you can select "advanced search" and just type `*substring*`. That works for me at least.
> I was looking for a way to match against a substring during the matching process for assigning tags, document types and correspondents. And this, I found out, can be achieved by just entering the substring in the match field and selecting the `regular expression` instead of `any` above.

Thanks!
So let's say i want to assign a tag when matching regular expression for "example" string. I can just select Regular Extension and put "example" as string to search?

I suppose it is more complex if i want to search multiple strings. So in this case i should use (/example | another_example/)?

There is a thing i do not understand, tried to match a string (that is fully searched/selectable manually on the pdf) with "Any matching", but the string is not found. So checking the pdf content in Paperless i found that the string is not isolated/formatted correctly and often it's combined with other strings. So i think the problem is there.

Basically the question is, if i have the word fully searchable with a CTLR-F in the pdf, but in the content in Paperless that work is not "isolated" i need to use Regular Expression?

**Reply by amo13** [reply]
I don't know. Why don't you just try it and find out?

Just to tell things apart: _Matching_ for automated tagging is different from _searching_ through your documents. I think we are talking about matching here.

My guess would be that the string you wish to match must figure somewhere in the OCR result text. If you use the regular expression for matching, the string would not need to be isolated and you could match "meth" in a document where the recognized content contains the word "something". 
For matching _"foo" or "bar"_ by regular expression, you would enter `foo|bar` in the match field for the tag. Note that there is no space because the regex would want to match the space as part of the string.

**Reply by steakhutzeee** [reply]
Thank you for the clarification!
